scripts/coccinelle/reserved_function_names.cocci

```diff
@@ -0,0 +1,98 @@
+/// Find function names starting with '__' or '_' and rename them
+// Copyright: (C) 2018 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+virtual patch
+virtual report
+
+@r_anyfn disable optional_storage@
+identifier fn =~ "^_.*";
+position p;
+@@
+
+  fn@p (...)
+
+@r_staticfn disable optional_storage@
+identifier fn =~ "^_.*";
+position p;
+@@
+
+  static fn@p (...) {...}
+
+@script:python depends on report && !r_staticfn@
+fn << r_anyfn.fn;
+pos << r_anyfn.p;
+@@
+msg="WARNING: Reserved function name " + fn
+coccilib.report.print_report(pos[0], msg)
+
+@script:python depends on report@
+fn << r_staticfn.fn;
+pos << r_staticfn.p;
+@@
+msg="WARNING: Reserved static function name " + fn
+coccilib.report.print_report(pos[0], msg)
+
+@script:python r_static_rewrite depends on patch@
+fn << r_staticfn.fn;
+z;
+@@
+
+r = fn
+
+while r.startswith('_'):
+    r = r[1:]
+
+coccinelle.z = r
+
+@script:python r_any_rewrite depends on patch@
+fn << r_anyfn.fn;
+z;
+@@
+
```

Comment:
Add here:
```
if fn == '__ASSERT':
          cocci.include_match(False)
```
There is one other variant I found `__ASSERT_NO_MSG`.
Maybe using regex would suffice but can't say, not a python
expert.

Otherwise you get :-/
```
diff -u -p a/pinmux_cc2650.c b/pinmux_cc2650.c
--- a/pinmux_cc2650.c
+++ b/pinmux_cc2650.c
@@ -96,7 +96,7 @@ static int pinmux_cc2650_get(struct devi
 static int pinmux_cc2650_pullup(struct device *dev, u32_t pin,
 				 u8_t func)
 {
-	__ASSERT((func == PINMUX_PULLUP_ENABLE) |
+	z_ASSERT((func == PINMUX_PULLUP_ENABLE) |
 		 (func == PINMUX_PULLUP_DISABLE),
 		 "Pullup mode is invalid");
 
@@ -117,7 +117,7 @@ static int pinmux_cc2650_pullup(struct d
 static int pinmux_cc2650_input(struct device *dev, u32_t pin,
 				u8_t func)
 {
-	__ASSERT((func == PINMUX_INPUT_ENABLED) |
+	z_ASSERT((func == PINMUX_INPUT_ENABLED) |
 		 (func == PINMUX_OUTPUT_ENABLED),
 		 "I/O mode is invalid");
```

Reply:
this should be Z_ASSERT, if you not match these cases you'll not be able to do the change.

Reply:
Z_ASSERT added to the list of prefixes in r_any_rewrite.